### ISSUE TITLE
Storybook is not working when using React Query

### ISSUE THREAD

**Issue by jvanbaarsen** [redwoodjs/graphql#2359]
When you setup React Query according to the docs you run into a couple of problems, one of them is that Storybook is no longer working:

![image](https://user-images.githubusercontent.com/1362793/116238104-42d73b00-a761-11eb-8c35-9de62cca5fd8.png)

**Comment by peterp** [comment]
@jvanbaarsen I'm using react query and storybook, are you using the package of here: https://github.com/MarceloAlves/redwoodjs-react-query-provider/

**Comment by jvanbaarsen** [comment]
@peterp Yeah that exact package. I needed to do some things to make testing work though see: #2298. Can that be the thing that is causing me problems?

**Comment by peterp** [comment]
@jvanbaarsen aaah, interesting. I'm using Cell's so this was completely abstracted away from me - are you directly interacting with the cache or the query/ mutation in some ways?

This could very likely be the cause of some very unexpected results.

**Comment by peterp** [comment]
We definitely need to fix this.

**Comment by jvanbaarsen** [comment]
@peterp Yeah I am. I need to mutate the cache directly after a mutation, otherwise, the user is seeing outdated data.

**Comment by jtoar** [comment]
@thedavidprice the next step here is: can a user hook into our storybook config to get this working manually? Or is our graphql client too coupled to storybook?

**Comment by thedavidprice** [comment]
Without digging into this more deeply, my understanding is that a user could handle this with a custom Storybook preview (overwriting the Redwood version from the testing package). However, as step one we need to resolve #2298

I'd prioritize accordingly:
- #2298 as P2/3 for v1
- this PR as `next`

**Comment by virtuoushub** [comment]
Anyone have a repo that can reproduce this issue for me to take a look at?
